File: ai-clinic-ui/scripts/validate-env.ts
// @ts-check
/**
 * Validate environment variables before build/commit.
 * Run: npx tsx scripts/validate-env.ts
 */

import { z } from "zod";
import * as dotenv from "dotenv";
import { resolve } from "path";

// Load .env file
dotenv.config({ path: resolve(process.cwd(), ".env") });

const envSchema = z.object({
  DATABASE_URL: z
    .string({ message: "DATABASE_URL is required" })
    .url("DATABASE_URL must be a valid URL"),
  NEXT_PUBLIC_API_URL: z
    .string({ message: "NEXT_PUBLIC_API_URL is required" })
    .url("NEXT_PUBLIC_API_URL must be a valid URL"),
});

function validateEnv(): boolean {
  console.log("🔍 Validating UI environment...");

  const result = envSchema.safeParse(process.env);

  if (!result.success) {
    console.log("❌ Environment validation failed:");
    result.error.issues.forEach((issue) => {
      console.log(`   - ${issue.path.join(".")}: ${issue.message}`);
    });
    return false;
  }

  console.log("✅ UI environment is valid");
  console.log(
    `   - DATABASE_URL: ${result.data.DATABASE_URL.replace(/:[^:@]+@/, ":****@")}`,
  );
  console.log(`   - NEXT_PUBLIC_API_URL: ${result.data.NEXT_PUBLIC_API_URL}`);
  return true;
}

const success = validateEnv();
process.exit(success ? 0 : 1);

File: ai-clinic-ocr/scripts/validate_env.py
#!/usr/bin/env python3
"""
Validate environment variables before commit/deploy.
Run: python scripts/validate_env.py
"""

import sys
from pathlib import Path

# Add parent directory to path for imports
sys.path.insert(0, str(Path(__file__).parent.parent))

from pydantic import ValidationError
from config import Settings


def validate_env() -> bool:
    """Validate all required environment variables are set."""
    print("🔍 Validating OCR service environment...")
    
    try:
        settings = Settings()
        
        errors = []
        
        # Check required fields
        if not settings.GEMINI_API_KEY.get_secret_value():
            errors.append("GEMINI_API_KEY is required but not set")
        
        # Validate CORS origins format
        origins = settings.get_cors_origins()
        if not origins:
            errors.append("CORS_ORIGINS must have at least one origin")
        
        for origin in origins:
            if not origin.startswith(("http://", "https://")):
                errors.append(f"Invalid CORS origin format: {origin}")
        
        # Validate port range
        if not (1 <= settings.PORT <= 65535):
            errors.append(f"PORT must be between 1 and 65535, got {settings.PORT}")
        
        if errors:
            print("❌ Environment validation failed:")
            for error in errors:
                print(f"   - {error}")
            return False
        
        print("✅ OCR service environment is valid")
        print(f"   - GEMINI_MODEL: {settings.GEMINI_MODEL}")
        print(f"   - HOST: {settings.HOST}")
        print(f"   - PORT: {settings.PORT}")
        print(f"   - CORS_ORIGINS: {', '.join(origins)}")
        return True
        
    except ValidationError as e:
        print("❌ Pydantic validation error:")
        for error in e.errors():
            print(f"   - {error['loc'][0]}: {error['msg']}")
        return False
    except Exception as e:
        print(f"❌ Unexpected error: {e}")
        return False


if __name__ == "__main__":
    success = validate_env()
    sys.exit(0 if success else 1)
